NEWS.md

```diff
@@ -30,6 +30,12 @@
     in `vec_assign()`), two double dispatch were needed. Now it can be
     done with one double dispatch by calling `vec_cast()` directly.
 
+* `vec_ptype2()` is now more permissive with subclasses of data frames
+  and fall back to a bare data frame when the classes do not have a
+  common type (#981). This is for convenience, users should normalise
+  their inputs to a common data frame class manually to avoid the
+  warning (if applicable, the classes could also implement a common type).
```

Comment:
Can we be a bit more encouraging here that really we want package developers to define the behavior?

```suggestion
  warning. If applicable, package developers should implement a common type to define the behavior precisely.
```

Reply:
(after talking more with Lionel, maybe ignore this if the message is mainly for users)